<launch>
	<arg name="id"       default="right" />
	<arg name="freq"     default="400.0"/>
	<arg name="offset_x" default="0.0"/>
    <arg name="offset_y" default="0.0"/>
    <arg name="offset_z" default="0.0"/>
	<arg name="virtual_platform" default="true" />
	<arg name="model" default="$(find foot_platform_ros)/urdf/$(arg id)/foot_platform.create.xacro" />

	<group ns="$(arg id)_platform">
	  	
		<param name="robot_description" command="$(find xacro)/xacro --inorder $(arg model) platform_id:=$(arg id)"/>  
		
		<group if="$(arg virtual_platform)">
			<node name="virtual_platform_joint_publisher" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui">
				<param name="publish_frequency" value="$(arg freq)"/>
				<remap from="joint_states" to="platform_joint_publisher/joint_states_2" />
			</node>	
		    <node name="platform_joint_publisher" pkg="foot_platform_ros" type="footPlatformVirtual" output="screen">
				<param name="platformID" value="$(arg id)"/>  
				<remap from="joint_states" to="platform_joint_publisher/joint_states" />
			</node>
		</group>	  

		<group unless="$(arg virtual_platform)">
			<node name="platform_joint_publisher" pkg="foot_platform_ros" type="footStatePublisher" output="screen">
				<param name="platformID" value="$(arg id)"/>  
				<remap from="joint_states" to="platform_joint_publisher/joint_states" />
			</node>
		</group>

		<node name="platform_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
			<param name="publish_frequency" type="double" value="1000.0"/>
			<remap from="joint_states" to="platform_joint_publisher/joint_states" />
		</node>
	
	</group>

	<node pkg="tf2_ros" type="static_transform_publisher" name="world_to_$(arg id)_platform" args="$(arg offset_x) $(arg offset_y) $(arg offset_z) 0 0 0 1 world $(arg id)_platform_base_link" />

</launch>
